perses/rjmc/geometry.py

```diff
@@ -291,8 +291,9 @@ def _logp_propose(self, top_proposal, old_positions, beta, new_positions=None, d
                 logZ_r = np.log((np.sqrt(2*np.pi)*(sigma_r/units.angstroms))) # CHECK DOMAIN AND UNITS
                 logp_r = self._bond_logq(r, bond, beta) - logZ_r
             else:
-                constraint = self._get_bond_constraint(atom, bond_atom, top_proposal.new_system)
-                r = constraint #set bond length to exactly constraint
+                if direction == 'forward':
+                    constraint = self._get_bond_constraint(atom, bond_atom, top_proposal.new_system)
```

Comment:
>I cant guess from the code what exactly _get_bond_constraint would return. Should you maybe still throw a common sense exception, in case the constraint is also None?

That could make sense. The system is pretty messed up if this is the case.

>I'm mostly worried about bugs slipping through in case a bond as well as constraint are absent.

If a topological bond is present, but there are no bond parameters or constraints, then yeah, that should fail. But it should be more graceful.